### ISSUE TITLE
bipip_node creation error "msg": "No JSON object could be decoded" when node name contains %

### ISSUE THREAD

**Issue by adamfitz** [F5Networks/f5-ansible#1381]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

bigip_node

##### ANSIBLE VERSION

```
ansible 2.8.0
  config file = /home/adam/pool/ansible.cfg
  configured module search path = [u'/home/adam/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Apr  9 2019, 14:30:50) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]

```

##### PYTHON VERSION

```
Python 2.7.5
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     12.1.3.4
  Build       0.0.2
  Edition     Point Release 4
  Date        Tue Apr 10 14:26:02 PDT 2018

```

##### ROLE VERSION

```
Role: f5devcentral.f5ansible
        description: Ansible role that allows use of the F5 Ansible Modules
        active: True
        commit: 2a327bd27251dfcb2da3eff68827bcd675b616a4
        commit_message: Merging upstream changes - 2019-06-13T12:57:14Z
        commit_url: https://api.github.com/repos/f5devcentral/ansible-role-f5ansible/git/commits/2a327bd27251dfcb2da3eff68827bcd675b616a4
        company: F5 Networks
        created: 2018-10-04T16:14:10.510085Z
        dependencies: []
        download_count: 6970
        forks_count: 0
        galaxy_info:
                author: caphrim007
                company: F5 Networks
                galaxy_tags: ['f5', 'networking', 'bigip', 'bigiq']
                license: GPLv3
                min_ansible_version: 2.8
                platforms: [{'name': 'EL', 'versions': ['all']}, {'name': 'Fedora', 'versions': ['all']}, {'name': 'Ubuntu', 'versions': ['all']}, {'name': 'D
                role_name: f5ansible
        github_branch: master
        github_repo: ansible-role-f5ansible
        github_user: f5devcentral
        id: 30341
        imported: 2019-06-13T09:01:54.430041-04:00
        install_date: Fri Jun 14 05:40:35 2019
        installed_version: v2019.6.7
        is_valid: True
        issue_tracker_url:
        license: GPLv3
        min_ansible_version: 2.8
        modified: 2019-06-13T13:01:54.437199Z
        open_issues_count: 0
        path: (u'/home/adam/.ansible/roles', u'/usr/share/ansible/roles', u'/etc/ansible/roles')
        role_type: ANS
        stargazers_count: 4
        travis_status_url:


```

##### CONFIGURATION

Ansible.cfg:
[defaults]
inventory = ./inventory/hosts

##### OS / ENVIRONMENT

CentOS Linux release 7.6.1810 (Core)

##### SUMMARY

Using the bigip_node module, when the node name contains a % character the task to create a node fails, but the node is actually created.  The use case it when using the route domain descriptor as part of the node name (eg: node_ip%1)

##### STEPS TO REPRODUCE

```
Playbook Command:
ansible-playbook ./create_pool_node.yml --extra-var "var_pool_name=test_pool var_partition=Corporate var_route_domain=1 var_lb_mode=round-robin var_pool_member_state=enabled var_listening_port=80"

Playbook:
---
- name: F5 bigip Pool creation playbook
  hosts: all
  gather_facts: no
  connection: local
  vars: # this is a lab box and test account
   username: <user name>
   password: <password>
  
  roles: 
    - f5devcentral.f5ansible

  tasks:
    - name: Create Nodes
      bigip_node:
        address: "{{ item.value }}"
        name: "{{ item.key + '%' + var_route_domain }}"
        state: present
        partition: Common
        fqdn_auto_populate: no 
        provider:
          server: "{{ inventory_hostname }}"
          user: "{{ username }}"
          password: "{{ password }}"
          validate_certs: no
      delegate_to: localhost
      loop: "{{ var_node_dict | dict2items}}"

    - name: Create Pool -> "{{ var_pool_name }}"
      bigip_pool:
        state: present
        name: "{{ var_pool_name }}"
        partition: "{{ var_partition}}" # pass the partition name at runtime
        lb_method: "{{ var_lb_mode }}" # pass the LB method at runtime
        provider:
          server: "{{ inventory_hostname }}"
          user: "{{ usernanme }}"
          password: "{{ password }}"
          validate_certs: no
      delegate_to: localhost

    - name: Create pool members in pool -> "{{ var_pool_name }}"
      bigip_pool_member:
        pool: "{{ var_pool_name }}"
        partition: "{{ var_partition}}"
        port: "{{ var_listening_port }}"
        state: "{{ var_pool_member_state }}"
        provider:
          server: "{{ inventory_hostname }}"
          user: "{{ usernanme }}"
          password: "{{ password }}"
          validate_certs: no
        address: item.value
        name: item.name
      delegate_to: localhost
      loop: "{{ var_pool_member_dict | dict2items}}"

all.yml file contents:

var_node_dict:
 172.24.131.10: 172.24.131.10%1
 172.24.131.8: 172.24.131.8%1
 172.24.131.9: 172.24.131.9%1

```

##### EXPECTED RESULTS

Nodes should be created and the rest of the playbook to continue execution.

##### ACTUAL RESULTS

```
[adam@ansible pool]$ ansible-playbook ./create_pool_node.yml --extra-var "var_pool_name=test_pool var_partition=Corporate var_route_domain=1 var_lb_mode=round-robin var_pool_member_state=enabled var_listening_port=80" -vvv
ansible-playbook 2.8.0
  config file = /home/adam/pool/ansible.cfg
  configured module search path = [u'/home/adam/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, Apr  9 2019, 14:30:50) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
Using /home/adam/pool/ansible.cfg as config file
host_list declined parsing /home/adam/pool/inventory/hosts as it did not pass it's verify_file() method
script declined parsing /home/adam/pool/inventory/hosts as it did not pass it's verify_file() method
auto declined parsing /home/adam/pool/inventory/hosts as it did not pass it's verify_file() method
Parsed /home/adam/pool/inventory/hosts inventory source with ini plugin

PLAYBOOK: create_pool_node.yml *******************************************************************************************************************************
1 plays in ./create_pool_node.yml

PLAY [F5 bigip Pool creation playbook] ***********************************************************************************************************************
META: ran handlers

TASK [f5devcentral.f5ansible : Check ansible version] ********************************************************************************************************
task path: /home/adam/.ansible/roles/f5devcentral.f5ansible/tasks/main.yaml:2
skipping: [lab2300b] => {}

TASK [Create Nodes] ******************************************************************************************************************************************
task path: /home/adam/pool/create_pool_node.yml:14
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: adam
<localhost> EXEC /bin/sh -c 'echo ~adam && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/adam/.ansible/tmp/ansible-tmp-1560491607.92-228816515260202 `" && echo ansible-tmp-1560491607.92-228816515260202="` echo /home/adam/.ansible/tmp/ansible-tmp-1560491607.92-228816515260202 `" ) && sleep 0'
Using module file /home/adam/.ansible/roles/f5devcentral.f5ansible/library/bigip_node.py
<localhost> PUT /home/adam/.ansible/tmp/ansible-local-20948oZGNln/tmpoDdn1W TO /home/adam/.ansible/tmp/ansible-tmp-1560491607.92-228816515260202/AnsiballZ_bigip_node.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/adam/.ansible/tmp/ansible-tmp-1560491607.92-228816515260202/ /home/adam/.ansible/tmp/ansible-tmp-1560491607.92-228816515260202/AnsiballZ_bigip_node.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python2 /home/adam/.ansible/tmp/ansible-tmp-1560491607.92-228816515260202/AnsiballZ_bigip_node.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /home/adam/.ansible/tmp/ansible-tmp-1560491607.92-228816515260202/ > /dev/null 2>&1 && sleep 0'
<localhost> EXEC /bin/sh -c 'echo ~adam && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_bigip_node_payload_0njmQk/__main__.py", line 1187, in main
    results = mm.exec_module()
  File "/tmp/ansible_bigip_node_payload_0njmQk/__main__.py", line 841, in exec_module
    changed = self.present()
  File "/tmp/ansible_bigip_node_payload_0njmQk/__main__.py", line 857, in present
    return self.create()
  File "/tmp/ansible_bigip_node_payload_0njmQk/__main__.py", line 941, in create
    self.create_on_device()
  File "/tmp/ansible_bigip_node_payload_0njmQk/__main__.py", line 1088, in create_on_device
    self._wait_for_fqdn_checks()
  File "/tmp/ansible_bigip_node_payload_0njmQk/__main__.py", line 1092, in _wait_for_fqdn_checks
    have = self.read_current_from_device()
  File "/tmp/ansible_bigip_node_payload_0njmQk/__main__.py", line 1003, in read_current_from_device
    raise F5ModuleError(str(ex))

failed: [lab2300b] (item={'key': u'172.24.131.10', 'value': u'172.24.131.10%1'}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "address": "172.24.131.10%1",
            "auth_provider": null,
            "availability_requirements": null,
            "connection_limit": null,
            "description": null,
            "dynamic_ratio": null,
            "fqdn": null,
            "fqdn_address_type": null,
            "fqdn_auto_populate": false,
            "fqdn_down_interval": null,
            "fqdn_up_interval": null,
            "monitor_type": null,
            "monitors": null,
            "name": "172.24.131.10%1",
            "partition": "Common",
            "password": null,
            "provider": {
                "auth_provider": null,
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
                "proxy_to": null,
                "server": "lab2300b",
                "server_port": null,
                "ssh_keyfile": null,
                "timeout": null,
                "transport": "rest",
                "user": "f5_script_user",
                "validate_certs": false
            },
            "quorum": null,
            "rate_limit": null,
            "ratio": null,
            "server": null,
            "server_port": null,
            "state": "present",
            "transport": null,
            "user": null,
            "validate_certs": null
        }
    },
    "item": {
        "key": "172.24.131.10",
        "value": "172.24.131.10%1"
    },
    "msg": "No JSON object could be decoded"
}
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/adam/.ansible/tmp/ansible-tmp-1560491610.28-230523620171202 `" && echo ansible-tmp-1560491610.28-230523620171202="` echo /home/adam/.ansible/tmp/ansible-tmp-1560491610.28-230523620171202 `" ) && sleep 0'
Using module file /home/adam/.ansible/roles/f5devcentral.f5ansible/library/bigip_node.py
<localhost> PUT /home/adam/.ansible/tmp/ansible-local-20948oZGNln/tmpbZySPU TO /home/adam/.ansible/tmp/ansible-tmp-1560491610.28-230523620171202/AnsiballZ_bigip_node.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/adam/.ansible/tmp/ansible-tmp-1560491610.28-230523620171202/ /home/adam/.ansible/tmp/ansible-tmp-1560491610.28-230523620171202/AnsiballZ_bigip_node.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python2 /home/adam/.ansible/tmp/ansible-tmp-1560491610.28-230523620171202/AnsiballZ_bigip_node.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /home/adam/.ansible/tmp/ansible-tmp-1560491610.28-230523620171202/ > /dev/null 2>&1 && sleep 0'
<localhost> EXEC /bin/sh -c 'echo ~adam && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_bigip_node_payload_dgjzOQ/__main__.py", line 1187, in main
    results = mm.exec_module()
  File "/tmp/ansible_bigip_node_payload_dgjzOQ/__main__.py", line 841, in exec_module
    changed = self.present()
  File "/tmp/ansible_bigip_node_payload_dgjzOQ/__main__.py", line 857, in present
    return self.create()
  File "/tmp/ansible_bigip_node_payload_dgjzOQ/__main__.py", line 941, in create
    self.create_on_device()
  File "/tmp/ansible_bigip_node_payload_dgjzOQ/__main__.py", line 1088, in create_on_device
    self._wait_for_fqdn_checks()
  File "/tmp/ansible_bigip_node_payload_dgjzOQ/__main__.py", line 1092, in _wait_for_fqdn_checks
    have = self.read_current_from_device()
  File "/tmp/ansible_bigip_node_payload_dgjzOQ/__main__.py", line 1003, in read_current_from_device
    raise F5ModuleError(str(ex))

failed: [lab2300b] (item={'key': u'172.24.131.9', 'value': u'172.24.131.9%1'}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "address": "172.24.131.9%1",
            "auth_provider": null,
            "availability_requirements": null,
            "connection_limit": null,
            "description": null,
            "dynamic_ratio": null,
            "fqdn": null,
            "fqdn_address_type": null,
            "fqdn_auto_populate": false,
            "fqdn_down_interval": null,
            "fqdn_up_interval": null,
            "monitor_type": null,
            "monitors": null,
            "name": "172.24.131.9%1",
            "partition": "Common",
            "password": null,
            "provider": {
                "auth_provider": null,
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
                "proxy_to": null,
                "server": "lab2300b",
                "server_port": null,
                "ssh_keyfile": null,
                "timeout": null,
                "transport": "rest",
                "user": "f5_script_user",
                "validate_certs": false
            },
            "quorum": null,
            "rate_limit": null,
            "ratio": null,
            "server": null,
            "server_port": null,
            "state": "present",
            "transport": null,
            "user": null,
            "validate_certs": null
        }
    },
    "item": {
        "key": "172.24.131.9",
        "value": "172.24.131.9%1"
    },
    "msg": "No JSON object could be decoded"
}
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/adam/.ansible/tmp/ansible-tmp-1560491612.42-42528664829563 `" && echo ansible-tmp-1560491612.42-42528664829563="` echo /home/adam/.ansible/tmp/ansible-tmp-1560491612.42-42528664829563 `" ) && sleep 0'
Using module file /home/adam/.ansible/roles/f5devcentral.f5ansible/library/bigip_node.py
<localhost> PUT /home/adam/.ansible/tmp/ansible-local-20948oZGNln/tmpMwYtFv TO /home/adam/.ansible/tmp/ansible-tmp-1560491612.42-42528664829563/AnsiballZ_bigip_node.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/adam/.ansible/tmp/ansible-tmp-1560491612.42-42528664829563/ /home/adam/.ansible/tmp/ansible-tmp-1560491612.42-42528664829563/AnsiballZ_bigip_node.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python2 /home/adam/.ansible/tmp/ansible-tmp-1560491612.42-42528664829563/AnsiballZ_bigip_node.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /home/adam/.ansible/tmp/ansible-tmp-1560491612.42-42528664829563/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_bigip_node_payload_1No40Z/__main__.py", line 1187, in main
    results = mm.exec_module()
  File "/tmp/ansible_bigip_node_payload_1No40Z/__main__.py", line 841, in exec_module
    changed = self.present()
  File "/tmp/ansible_bigip_node_payload_1No40Z/__main__.py", line 857, in present
    return self.create()
  File "/tmp/ansible_bigip_node_payload_1No40Z/__main__.py", line 941, in create
    self.create_on_device()
  File "/tmp/ansible_bigip_node_payload_1No40Z/__main__.py", line 1088, in create_on_device
    self._wait_for_fqdn_checks()
  File "/tmp/ansible_bigip_node_payload_1No40Z/__main__.py", line 1092, in _wait_for_fqdn_checks
    have = self.read_current_from_device()
  File "/tmp/ansible_bigip_node_payload_1No40Z/__main__.py", line 1003, in read_current_from_device
    raise F5ModuleError(str(ex))

failed: [lab2300b] (item={'key': u'172.24.131.8', 'value': u'172.24.131.8%1'}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "address": "172.24.131.8%1",
            "auth_provider": null,
            "availability_requirements": null,
            "connection_limit": null,
            "description": null,
            "dynamic_ratio": null,
            "fqdn": null,
            "fqdn_address_type": null,
            "fqdn_auto_populate": false,
            "fqdn_down_interval": null,
            "fqdn_up_interval": null,
            "monitor_type": null,
            "monitors": null,
            "name": "172.24.131.8%1",
            "partition": "Common",
            "password": null,
            "provider": {
                "auth_provider": null,
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
                "proxy_to": null,
                "server": "lab2300b",
                "server_port": null,
                "ssh_keyfile": null,
                "timeout": null,
                "transport": "rest",
                "user": "f5_script_user",
                "validate_certs": false
            },
            "quorum": null,
            "rate_limit": null,
            "ratio": null,
            "server": null,
            "server_port": null,
            "state": "present",
            "transport": null,
            "user": null,
            "validate_certs": null
        }
    },
    "item": {
        "key": "172.24.131.8",
        "value": "172.24.131.8%1"
    },
    "msg": "No JSON object could be decoded"
}

PLAY RECAP ***************************************************************************************************************************************************
lab2300b                : ok=0    changed=0    unreachable=0    failed=1    skipped=1    rescued=0    ignored=0

[adam@ansible pool]$


```

**Comment by wojtek0806** [comment]
@adamfitz 
the route domain needs to be applied to address not the name, names are just arbitrary labels. Also please check if this works with tmsh, if it does then we will address this, if tmsh does not allow % in node name when creating it then you have your answer :)

**Comment by adamfitz** [comment]
> @adamfitz
> the route domain needs to be applied to address not the name, names are just arbitrary labels. Also please check if this works with tmsh, if it does then we will address this, if tmsh does not allow % in node name when creating it then you have your answer :)

Hi,

Yep I understand the name is just the text label, this is our naming scheme as it is easy to identify the ip and route domain by just searching the node name.  The tmsh command accepts this, we use this convention for all nodes, example tmsh command output below:

```
adam@(lab2300b)(cfg-sync In Sync)(Active)(/Common)(tmos)# create ltm node 172.24.131.10%1 { address 172.24.131.10%1 }
adam@(lab2300b)(cfg-sync In Sync)(Active)(/Common)(tmos)# list ltm node 172.24.131.10%1
ltm node 172.24.131.10%1 {
    address 172.24.131.10%1
    session monitor-enabled
    state checking
}
```
Originally I thought this problem was that the module would not allow creation of a node with the % char.  But since all three nodes actually get created on the device before the task errors I think it might just be related to rendering of the return data which the error also seems to indicate.

**Comment by adamfitz** [comment]
In addition to manually testing the tmsh commands, I did a test with the bigip_command module which also worked so it seems like this issue is related specifically to the bigip_node module.

Test Playbook:
```
---
- name: Testing bigip_command module
  hosts: all
  gather_facts: no
  connection: local
  vars: # this is a lab box and test account
   username: username
   password: password
  
  roles: 
    - f5devcentral.f5ansible

  tasks:
    - name: Create Nodes - with BIGIP_COMMAND module
      bigip_command:
        commands:
          - create ltm node "{{ '/Common/' +  item.key + '%' + var_route_domain }}" address "{{ item.value }}"
        provider:
          server: "{{ inventory_hostname }}"
          user: "{{ username }}"
          password: "{{ password }}"
          validate_certs: no
      register: results
      delegate_to: localhost
      loop: "{{ var_node_dict | dict2items}}"
      loop_control:
        label: "Node creation attempt: {{ results }}"
```

all.yml:
```
var_node_dict:
 172.24.131.10: 172.24.131.10%1
 172.24.131.8: 172.24.131.8%1
 172.24.131.9: 172.24.131.9%1
```
Successful output:
```
[adam@ansible pool]$ ansible-playbook ./add_node_bigip_command.yml --extra-var "var_route_domain=1"

PLAY [Testing bigip_command module] **************************************************************************************************************

TASK [f5devcentral.f5ansible : Check ansible version] ********************************************************************************************
skipping: [lab2300b]

TASK [Create Nodes - with BIGIP_COMMAND module] **************************************************************************************************
changed: [lab2300b] => (item=Node creation attempt: {'_ansible_parsed': True, 'ansible_loop_var': 'item', u'stdout': [], '_ansible_no_log': False, '_ansible_delegated_vars': {'ansible_delegated_host': u'localhost', 'ansible_host': u'localhost'}, '_ansible_item_result': True, u'changed': True, 'item': {'value': u'172.24.131.10%1', 'key': u'172.24.131.10'}, u'warnings': [u'Using "write" commands is not idempotent. You should use a module that is specifically made for that. If such a module does not exist, then please file a bug. The command in question is "create ltm node "172.24.131.10%1" addres..."'], u'stdout_lines': [], u'invocation': {u'module_args': {u'retries': 10, u'commands': [u'create ltm node "172.24.131.10%1" address "172.24.131.10%1"'], u'chdir': None, u'auth_provider': None, u'server': None, u'interval': 1, u'warn': True, u'password': None, u'user': None, u'server_port': None, u'provider': {u'ssh_keyfile': None, u'server_port': None, u'auth_provider': None, u'proxy_to': None, u'server': u'lab2300b', u'user': u'f5_script_user', u'timeout': None, u'password': u'VALUE_SPECIFIED_IN_NO_LOG_PARAMETER', u'validate_certs': False, u'transport': u'rest'}, u'wait_for': None, u'validate_certs': None, u'transport': u'rest', u'match': u'all'}}, u'executed_commands': [u'tmsh -c \\"create ltm node \\\\\\"172.24.131.10%1\\\\\\" address \\\\\\"172.24.131.10%1\\\\\\"\\"'], '_ansible_ignore_errors': None, 'failed': False})
changed: [lab2300b] => (item=Node creation attempt: {'_ansible_parsed': True, 'ansible_loop_var': 'item', u'stdout': [], '_ansible_no_log': False, '_ansible_delegated_vars': {'ansible_delegated_host': u'localhost', 'ansible_host': u'localhost'}, '_ansible_item_result': True, u'changed': True, 'item': {'value': u'172.24.131.9%1', 'key': u'172.24.131.9'}, u'warnings': [u'Using "write" commands is not idempotent. You should use a module that is specifically made for that. If such a module does not exist, then please file a bug. The command in question is "create ltm node "172.24.131.9%1" address..."'], u'stdout_lines': [], u'invocation': {u'module_args': {u'retries': 10, u'commands': [u'create ltm node "172.24.131.9%1" address "172.24.131.9%1"'], u'chdir': None, u'auth_provider': None, u'server': None, u'interval': 1, u'warn': True, u'password': None, u'user': None, u'server_port': None, u'provider': {u'ssh_keyfile': None, u'server_port': None, u'auth_provider': None, u'proxy_to': None, u'server': u'lab2300b', u'user': u'f5_script_user', u'timeout': None, u'password': u'VALUE_SPECIFIED_IN_NO_LOG_PARAMETER', u'validate_certs': False, u'transport': u'rest'}, u'wait_for': None, u'validate_certs': None, u'transport': u'rest', u'match': u'all'}}, u'executed_commands': [u'tmsh -c \\"create ltm node \\\\\\"172.24.131.9%1\\\\\\" address \\\\\\"172.24.131.9%1\\\\\\"\\"'], '_ansible_ignore_errors': None, 'failed': False})
changed: [lab2300b] => (item=Node creation attempt: {'_ansible_parsed': True, 'ansible_loop_var': 'item', u'stdout': [], '_ansible_no_log': False, '_ansible_delegated_vars': {'ansible_delegated_host': u'localhost', 'ansible_host': u'localhost'}, '_ansible_item_result': True, u'changed': True, 'item': {'value': u'172.24.131.8%1', 'key': u'172.24.131.8'}, u'warnings': [u'Using "write" commands is not idempotent. You should use a module that is specifically made for that. If such a module does not exist, then please file a bug. The command in question is "create ltm node "172.24.131.8%1" address..."'], u'stdout_lines': [], u'invocation': {u'module_args': {u'retries': 10, u'commands': [u'create ltm node "172.24.131.8%1" address "172.24.131.8%1"'], u'chdir': None, u'auth_provider': None, u'server': None, u'interval': 1, u'warn': True, u'password': None, u'user': None, u'server_port': None, u'provider': {u'ssh_keyfile': None, u'server_port': None, u'auth_provider': None, u'proxy_to': None, u'server': u'lab2300b', u'user': u'f5_script_user', u'timeout': None, u'password': u'VALUE_SPECIFIED_IN_NO_LOG_PARAMETER', u'validate_certs': False, u'transport': u'rest'}, u'wait_for': None, u'validate_certs': None, u'transport': u'rest', u'match': u'all'}}, u'executed_commands': [u'tmsh -c \\"create ltm node \\\\\\"172.24.131.8%1\\\\\\" address \\\\\\"172.24.131.8%1\\\\\\"\\"'], '_ansible_ignore_errors': None, 'failed': False})
 [WARNING]: Using "write" commands is not idempotent. You should use a module that is specifically made for that. If such a module does not
exist, then please file a bug. The command in question is "create ltm node "172.24.131.10%1" addres..."

 [WARNING]: Using "write" commands is not idempotent. You should use a module that is specifically made for that. If such a module does not
exist, then please file a bug. The command in question is "create ltm node "172.24.131.9%1" address..."

 [WARNING]: Using "write" commands is not idempotent. You should use a module that is specifically made for that. If such a module does not
exist, then please file a bug. The command in question is "create ltm node "172.24.131.8%1" address..."


PLAY RECAP ***************************************************************************************************************************************
lab2300b                : ok=1    changed=1    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0

[adam@ansible pool]$
```

**Comment by wojtek0806** [comment]
FMFA-295

**Comment by wojtek0806** [comment]
@nitinthewiz i think this is also related to your fix, ill assign it to you can you work on it concurrently please?

**Comment by wojtek0806** [comment]
@nitinthewiz your fix for #1401 is merged, write integration tests for this issue and make sure they pass, if not then make changes in bigip_node module, submit PR separately please :)
